### PR TITLE
[stable/heapster] add rbac for heapster

### PR DESCRIPTION
**Pull request by smelchior**
picking up #1337 i added rbac for heapster
This adds an additional role for the pod nanny. 
I also renamed rbac.enabled to rbac.create as that is the flag most of the other charts tend to use

### CODE REVIEW

**Comment by rimusz** [comment]
PR name should really include chart name as `[stable/heapster] add rbac` as otherwise you get confused which chart is PR for.

**Comment by smelchior** [comment]
sorry, that does indeed make sense :)

File: stable/heapster/templates/role.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{{- if .Values.rbac.create -}}
+{{- if .Values.resizer.enabled -}}
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: Role
+metadata:
+  name: system:pod-nanny
```
Comment (mgoodness) [inline stable/heapster/templates/role.yaml]: Let's make this `{{ template "fullname" . }}-pod-nanny` and rename the file to `pod-nanny-role.yaml`.

File: stable/heapster/templates/rolebinding.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  name: {{ template "fullname" . }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:heapster
+subjects:
+  - kind: ServiceAccount
+    name: {{ template "fullname" . }}
+    namespace: {{ .Release.Namespace }}
+{{ if .Values.resizer.enabled -}}
```
Comment (mgoodness) [inline stable/heapster/templates/rolebinding.yaml]: Split these into individual files, `heapster-crb.yaml` and `pod-nanny-rolebinding.yaml`

File: stable/heapster/templates/rolebinding.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: RoleBinding
+metadata:
+  name: {{ template "fullname" . }}-nanny
```
Comment (mgoodness) [inline stable/heapster/templates/rolebinding.yaml]: `{{ template "fullname" . }}-pod-nanny`

File: stable/heapster/templates/rolebinding.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  labels:
+    app: {{ template "name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: system:pod-nanny
```
Comment (mgoodness) [inline stable/heapster/templates/rolebinding.yaml]: `{{ template "fullname" . }}-pod-nanny`

File: stable/heapster/templates/deployment.yaml [hunk 5]
```diff
@@ -14,6 +14,9 @@ spec:
       labels:
         app: {{ template "fullname" . }}
     spec:
+{{- if .Values.rbac.create }}
+      serviceAccountName: {{ template "fullname" . }}
```
Comment (mgoodness) [inline stable/heapster/templates/deployment.yaml]: Create an `rbac.serviceAccountName` and use `serviceAccountName: {{ if .Values.rbac.create }}{{ template "fullname" . }}{{ else }}"{{ .Values.rbac.serviceAccountName }}"{{ end }}`

**Review by mgoodness — CHANGES_REQUESTED** [review]
Looks pretty good. Just a few changes to finish it up. Don't forget to bump the chart version!

**Review by mgoodness — APPROVED** [review]

**Comment by smelchior** [comment]
thanks :)!

**Pull request merged**